libs/api/ebpf_api.cpp

```diff
@@ -64,16 +69,44 @@ create_map_function(
 
     _ebpf_operation_create_map_reply reply{};
 
-    uint32_t retval = invoke_ioctl(device_handle, request, reply);
-    if (retval != ERROR_SUCCESS) {
-        throw std::runtime_error(std::string("Error ") + std::to_string(retval) + " trying to create map");
+    uint32_t return_value = ERROR_SUCCESS;
+
+    if (handle == NULL) {
+        return_value = ERROR_INVALID_PARAMETER;
+        goto Exit;
     }
 
+    *handle = INVALID_HANDLE_VALUE;
+
+    return_value = invoke_ioctl(device_handle, request, reply);
+
+    if (return_value != ERROR_SUCCESS)
+        goto Exit;
+
     if (reply.header.id != ebpf_operation_id_t::EBPF_OPERATION_CREATE_MAP) {
-        throw std::runtime_error(std::string("reply.header.id != ebpf_operation_id_t::EBPF_OPERATION_CREATE_MAP"));
+        ebpf_assert(false);
```

Comment:
Perhaps a more specific ebpf_result_t can be returned which the calling function can then intepret and return a more useful error message. But the reply header type is something very Windows eBPF implementation specific and has got nothing to do with program verification. Also I don't see this error happening ever. Maybe we can remove this case altogether or simply put an assert. Thoughts?

Reply:
if it would be a bug in the ebpf framework implementation for this to be hit, then assert.
If it could be hit due to anything other than a bug in the ebpf framewrok implementation (e.g., permission denied, out of memory, or a bug in the calling application, then return an error)

Reply:
Yes it would be a bug in framework. So changed this to an assert instead.